jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/logging/CliLoggerTest.java

```diff
@@ -0,0 +1,195 @@
+    Mockito.verifyNoMoreInteractions(mockOut);
+  }
+
+  @Test
+  public void testLog_info() {
+    CliLogger logger = new CliLogger(Verbosity.info, mockOut, mockErr);
+    sendMessages(logger);
+    Mockito.verify(mockErr).println("error");
+    Mockito.verifyNoMoreInteractions(mockErr);
+    Mockito.verify(mockOut).println("warn");
+    Mockito.verify(mockOut).println("lifecycle");
+    Mockito.verify(mockOut).println("info");
+    Mockito.verifyNoMoreInteractions(mockOut);
+  }
+
+  @Test
+  public void testLog_debug() {
+    CliLogger logger = new CliLogger(Verbosity.debug, mockOut, mockErr);
+    sendMessages(logger);
+    Mockito.verify(mockErr).println("error");
+    Mockito.verifyNoMoreInteractions(mockErr);
+    Mockito.verify(mockOut).println("warn");
+    Mockito.verify(mockOut).println("lifecycle");
+    Mockito.verify(mockOut).println("info");
+    Mockito.verify(mockOut).println("debug");
+    Mockito.verifyNoMoreInteractions(mockOut);
+  }
+
+  @Test
+  public void testIsRichConsole_true() {
+    assertThat(CliLogger.isRichConsole(ConsoleOutput.rich)).isTrue();
+  }
+
+  @Test
+  public void testIsRichConsole_false() {
+    assertThat(CliLogger.isRichConsole(ConsoleOutput.plain)).isFalse();
+  }
+
+  @Test
+  public void testIsRightConsole_autoWindowsTrue() {
+    System.setProperty("os.name", "windows");
+    assertThat(CliLogger.isRichConsole(ConsoleOutput.auto)).isTrue();
+  }
+
+  @Test
+  public void testIsRightConsole_autoTermTrue() {
+    environmentVariables.set("TERM", "not-dumb");
+    assertThat(CliLogger.isRichConsole(ConsoleOutput.auto)).isTrue();
+  }
+
+  @Test
+  public void testIsRightConsole_autoDumbTermFalse() {
+    environmentVariables.set("TERM", "dumb");
+    assertThat(CliLogger.isRichConsole(ConsoleOutput.auto)).isFalse();
+  }
+
+  @Test
+  public void testNewLogger_richConfig() {
+    InOrder inOrder = Mockito.inOrder(mockCliLogger);
+    SingleThreadedExecutor singleThreadedExecutor = new SingleThreadedExecutor();
+    ConsoleLogger logger = CliLogger.newLogger(mockCliLogger, true, singleThreadedExecutor);
+
+    logger.log(LogEvent.Level.DEBUG, "debug");
+    logger.log(LogEvent.Level.INFO, "info");
+    logger.log(LogEvent.Level.LIFECYCLE, "lifecycle");
+    logger.log(LogEvent.Level.PROGRESS, "progress");
+    logger.log(LogEvent.Level.WARN, "warn");
+    logger.log(LogEvent.Level.ERROR, "error");
+
+    singleThreadedExecutor.shutDownAndAwaitTermination(Duration.ofSeconds(3));
+
+    inOrder.verify(mockCliLogger).debug("debug");
+    inOrder.verify(mockCliLogger).info("info");
+    inOrder.verify(mockCliLogger).lifecycle("lifecycle");
+    inOrder.verify(mockCliLogger).warn("warn");
+    inOrder.verify(mockCliLogger).error("error");
+    inOrder.verifyNoMoreInteractions(); // progress is not configured
+  }
+
+  @Test
+  public void testNewLogger_plainConfig() {
+    InOrder inOrder = Mockito.inOrder(mockCliLogger);
+    SingleThreadedExecutor singleThreadedExecutor = new SingleThreadedExecutor();
```

Comment:
BTW, doesn't `moreexecutors.directexecutor()` work?

Reply:
It would, but the type that the ConsoleLogger accepts is explicitly SingleThreadedExecutor. So kind of limited by that. We must have had a reason for doing this a while ago, maybe to help with testing and didn't know a better way?

Reply:
but it looks more like just to handle shutting down better.